MH4U Dex/CoreDataController.m

```diff
@@ -167,41 +168,51 @@ - (void)loadQuestDataWithContext:(NSManagedObjectContext *)context
         quest = (Quest *)[self uniqueEntityWithEntityName:[Quest entityName]
                                             withPredicate:questPredicate
                                               withContext:context];
-        //Only add the quest if it is not already there
+        //Even if the quest is already there, it may have been created as a dummy prerequisite quest, and may need more options filled out.
         if (!quest) {
-            // Quests have a lot of information to populate, so I'm moving this to a helper method to cut down on the size of this method.
-            quest = [self questFromDictionary:questDict withContext:context];
+            quest = [Quest insertInManagedObjectContext:context];
         }
-    }
-    if (![self attemptSaveContext:context]) {
-        NSLog(@"WARNING: Failed to save the item context!");
+        [self updateQuest:quest
+           fromDictionary:questDict
+              withContext:context];
     }
 }
 
 - (void)loadQuestDropDataWithContext:(NSManagedObjectContext *)context
 {
-    NSArray *questDropList = [self loadArrayFromJsonFileNamed:MHDQuestDropsFileName];
-    
-    for (NSDictionary *dropDict in questDropList) {
-        NSNumber *dropID = dropDict[MHDQuestDropIDKey];
-        NSPredicate *dropPredicate = [NSPredicate predicateWithFormat:@"%K == %@", QuestDropAttributes.id, dropID];
-        QuestDrop *drop = (QuestDrop *)[self uniqueEntityWithEntityName:[QuestDrop entityName]
-                                                          withPredicate:dropPredicate
-                                                            withContext:context];
-        if (!drop) {
-            //Returns nil if the quest corresponding to this drop does not exist.
-            drop = [self questDropFromDictionary:dropDict WithContext:context];
-            if (!drop) {
-                //If we reach this point, then the quest that this drop belongs to did not exist, and we should return without saving the context;
-                //TODO: Better way to handle this?  Though this never SHOULD happen.
-                return;
+    //Repeat this process for each JSON file
+    NSArray *urlArray = [[NSBundle mainBundle] URLsForResourcesWithExtension:MHDJSON subdirectory:nil];
+    for (NSURL *url in urlArray)  {
+        NSString *fileName = [[url URLByDeletingPathExtension] lastPathComponent];
+        if ([fileName hasPrefix:QuestDropFilePrefix]) {
+            NSLog(@"Loading in quest drop file: %@", fileName);
```

Comment:
Yes, for now it helps me figure out if I've stalled loading a single file or not.